express-app/controllers/section.js

```diff
@@ -17,43 +21,45 @@ module.exports = {
    * @param {object} request - request object
    * @param {object} response - response object.
    */
-  getQuestions: function (request, response) {
+  getQuestions: function (user, request, response) {
     // Call promise that validates internet connection
-    isOnline().then((online) => {
-      if (online) {
-        req.get(
-          // url to get
-          urls.apiUrl + urls.api.questions,
-          {
-            headers: {
-              'Authorization': 'Token ' + request.session.apiToken,
+    if (!fs.existsSync(sectionsPath)) {
+      isOnline().then((online) => {
```

Comment:
Why do you check if the user is online again? This is already handled at the user controller.